components/scream/scripts/test_all_scream.py

```diff
@@ -85,7 +91,11 @@ def __init__(self, cxx_compiler, f90_compiler, c_compiler, submit=False, paralle
                                   "fpe" : "debug_nopack_fpe"}
 
         if not self._tests:
-            self._tests = ["dbg", "sp", "fpe"]
+            is_cuda_machine = "OMPI_CXX" in os.environ
```

Comment:
I think we should not rely on OMPI_CXX. Ekat can automatically set that at config time, so one may not set it in the env. Even if now it is in the machine specs, if one adds a new GPU machine, he should not need to add OMPI_CXX to the env (since ekat auto sets that for you). Besides, OMPI_CXX is *only* good for openmpi. MPICH has a different env var (which ekat sets correctly).

Perhaps a better way to do this is to check if `nvcc` is present in the env PATH. I would assume nobody loads cuda when building a non-gpu version of scream...